sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/run/android/SqlDelightRunAnnotator.kt

```diff
@@ -0,0 +1,78 @@
+package app.cash.sqldelight.intellij.run.android
+
+import app.cash.sqldelight.core.lang.util.rawSqlText
+import com.alecstrong.sql.psi.core.psi.SqlStmt
+import com.alecstrong.sql.psi.core.psi.SqlStmtList
+import com.android.tools.idea.sqlite.DatabaseInspectorProjectService
+import com.android.tools.idea.sqlite.annotator.RunSqliteStatementGutterIconAction
+import com.android.tools.idea.sqlite.ui.DatabaseInspectorViewsFactoryImpl
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.lang.java.JavaLanguage
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.PsiLanguageInjectionHost
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.ui.EmptyIcon
+import icons.StudioIcons
+import javax.swing.Icon
+
+class SqlDelightRunAnnotator : Annotator {
+  override fun annotate(element: PsiElement, holder: AnnotationHolder) {
+    if (element.parent !is SqlStmtList || element !is SqlStmt) return
+
+    val text = element.rawSqlText().trim().replace("\\s+".toRegex(), " ")
+    val s = """
+      |package com.example;
+      |
+      |import android.database.sqlite.SQLiteDatabase;
+      |
+      |class Util {
+      |    void f(SQLiteDatabase db) {
+      |        db.execSQL("$text");
```

Comment:
Can this whole operation be made lazy to run when you actually click? Currently it seems like it would have to run on every keystroke to stay up-to-date.

Reply:
yea totally, let me try that